src/player-area.js

```diff
@@ -239,36 +214,50 @@ export const PlayerArea = () => {
 
           centerMap(true);
           setShowSplashScreen(false);
+        })
+        .catch(err => {
+          // @TODO: distinguish between network error (rertry?) and cancel error
+          console.error(err);
         });
-      }
+    });
 
-      currentMapId.current = data.mapId;
+    const contextmenuListener = ev => {
+      ev.preventDefault();
+    };
+    window.addEventListener("contextmenu", contextmenuListener);
 
-      const contextmenuListener = ev => {
-        ev.preventDefault();
-      };
-      window.addEventListener("contextmenu", contextmenuListener);
+    return () => {
+      window.removeEventListener("contextmenu", contextmenuListener);
 
-      return () => {
-        window.removeEventListener("contextmenu", contextmenuListener);
-      };
-    });
+      if (pendingImageLoads.current) {
+        pendingImageLoads.current.forEach(task => {
+          task.cancel();
+        });
+        pendingImageLoads.current = null;
+      }
+
+      socket.close();
+    };
   }, []);
 
+  /**
+   * long press event for setting a map marker
+   */
   useLongPress(ev => {
     if (!mapCanvasDimensions.current) {
       return;
     }
 
     let input = null;
-    // ev can be touch or click event
+    // ev can be a touch or click event
     if (ev.touches) {
       input = [ev.touches[0].pageX, ev.touches[0].pageY];
     } else {
       input = [ev.pageX, ev.pageY];
     }
+
     // calculate coordinates relative to the canvas
-    const ref = new Referentiel(panZoomRef.current.dragContainer);
+    const ref = new Referentiel(panZoomRef.current.getDragContainer());
```

Comment:
usage of this method is currently unsafe since it is not documented on react-native-easy-pan-zoom yet. https://github.com/mnogueron/react-easy-panzoom/pull/10